src/include/common/object_pool.h

```diff
@@ -76,17 +104,72 @@ class ObjectPool {
   /**
    * Returns a piece of memory to hold an object of T.
    *
+   * It throws exception if the object pool fails to fetch memory.
+   *
    * @return pointer to memory that can hold T
    */
   T *Get() {
     T *result = nullptr;
-    if (!reuse_queue_.Dequeue(&result))
-      result = alloc_.New();
-    else
+    if (!reuse_queue_.Dequeue(&result)) {
+      curr_add_lock_.Lock();
+      size_limit_lock_.Lock();
```

Comment:
Use ScopedSpinLatch for these

Reply:
I don't think ScopedSpinLatch fit in this case. The latches are members in an object pool. If we use ScopedSpinLatch, then latches will only be unlocked when the object pool is destroyed, which is not something we want.